src/OrchardCore.Modules/OrchardCore.AutoSetup/Options/TenantSetupOptions.cs

```diff
@@ -0,0 +1,147 @@
+
+        /// <summary>
+        /// Gets or sets the user friendly Tenant/Site Name.
+        /// </summary>
+        public string SiteName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the admin username.
+        /// </summary>
+        public string AdminUsername { get; set; }
+
+        /// <summary>
+        /// Gets or sets the admin email.
+        /// </summary>
+        public string AdminEmail { get; set; }
+
+        /// <summary>
+        /// Gets or sets the admin password.
+        /// </summary>
+        public string AdminPassword { get; set; }
+
+        /// <summary>
+        /// Gets or sets the database provider.
+        /// </summary>
+        public string DatabaseProvider { get; set; }
+
+        /// <summary>
+        /// Gets or sets the database connection string.
+        /// </summary>
+        public string DatabaseConnectionString { get; set; }
+
+        /// <summary>
+        /// Gets or sets the database table prefix.
+        /// </summary>
+        public string DatabaseTablePrefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets the recipe name.
+        /// </summary>
+        public string RecipeName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the site time zone.
+        /// </summary>
+        public string SiteTimeZone { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tenants request url host.
+        /// </summary>
+        public string RequestUrlHost { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tenant request url prefix.
+        /// </summary>
+        public string RequestUrlPrefix { get; set; }
+
+        /// <summary>
+        /// Gets the Flag which indicates a Default/Root shell/tenant.
+        /// </summary>
+        public bool IsDefault => string.Equals(ShellName, DefaultShellName, StringComparison.InvariantCultureIgnoreCase);
+
+        /// <summary>
+        /// The validate.
+        /// </summary>
+        /// <param name="validationContext"> The validation context. </param>
+        /// <returns>
+        /// The collection of errors if any. </returns>
+        public virtual IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            var T = validationContext.GetService<IStringLocalizer<AutoSetupOptions>>();
+
+            if (String.IsNullOrEmpty(ShellName) || !Regex.IsMatch(ShellName, @"^\w+$"))
+            {
+                yield return new ValidationResult(T["Invalid shell name. Can not be empty and must contain characters only and no spaces.", "Shell name"], new[] { nameof(ShellName) });
+            }
+
+            if (!IsDefault && string.IsNullOrWhiteSpace(RequestUrlPrefix) && string.IsNullOrWhiteSpace(RequestUrlHost))
+            {
+                yield return new ValidationResult(T["For no Default Tenant RequestUrlPrefix or RequestUrlHost should be provided", "Tenant Url"], new[] { "TenantUrl" });
+            }
```

Comment:
I didn't check all properties, only the `RequestUrlPrefix`. So here maybe good to add the following check as it is done in the Tenants controller and api controller, but anyway not in the WF CreateTenantTask, so as you want ;)

Note: Validations would need to be consistent everywhere, maybe in a separate PR.


            if (!String.IsNullOrWhiteSpace(shellSettings.RequestUrlPrefix))
            {
                if (shellSettings.RequestUrlPrefix.Contains('/'))
                {
                    ModelState.AddModelError(nameof(CreateApiViewModel.RequestUrlPrefix), S["The url prefix can not contain more than one segment."]);
                }
            }